openedx/core/lib/api/authentication.py

```diff
@@ -128,3 +133,27 @@ def authenticate_credentials(self, request, access_token):
             })
         else:
             return token.user, token
+
+
+class JwtAuthentication(JSONWebTokenAuthentication):
+    """
+    Custom authentication using JWT.
+    """
+
+    def authenticate_credentials(self, payload):
+        """
+        Return a user object to be associated with the present request, based on
+        the content of an already-decoded / verified JWT payload.
+        """
+        if 'preferred_username' not in payload:
```

Comment:
Why are we using `preferred_username`? That is an OpenID Connect concept. We use `username` for all of our APIs. If you do the same, this custom class is not necessary.